recipes/libpq/all/conanfile.py

```diff
@@ -174,6 +174,14 @@ def _patch_sources(self):
                 replace_in_file(self,config_default_pl,
                                       openssl_entry,
                                       "openssl   => '%s'" % openssl.package_folder.replace("\\", "/"))
+            if "zlib" in self.dependencies.host:
+                zlib = self.dependencies["zlib"]
+                replace_in_file(self, solution_pm, "zdll.lib", f"{zlib.cpp_info.libs[0]}.lib")
+                zlib_entry = "zlib => undef" if Version(self.version) >= "16.0" else "zlib      => undef"
+                replace_in_file(self, config_default_pl,
+                                zlib_entry,
+                                "zlib   => '%s'" % zlib.package_folder.replace("\\", "/"))
```

Comment:
this has side effects

If I'm reading the config file correctly

```
	$cfg .= ' --without-zlib' unless ($self->{options}->{zlib});
```

The default is `--without-zlib`
If we are providing zlib via the option (it now has a value where it didn't before), then that would cause `HAVE_LIBZ` to be defined, and things to be built in the implementation that were not before (if Im not mistaken this recipe has never attempted to build with zlib support or link to it, and the non-msvc code paths variant explicitly disable it).